tools/driver/CMakeLists.txt

```diff
@@ -38,6 +38,11 @@ if(SWIFT_HOST_VARIANT MATCHES "${SWIFT_DARWIN_VARIANTS}")
       LINK_FLAGS " -lobjc -Wl,-force_load,\"${SWIFT_REPL_ARCLITE}\"")
 endif()
 
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "CYGWIN")
+  set_property(TARGET swift APPEND_STRING PROPERTY
+      LINK_FLAGS " -Wl,--allow-multiple-definition")
```

Comment:
The GNU ld linker has a very poor approximation of COFF support.  It fails constantly under a myriad of conditions, including COMDAT symbols.  The `--allow-multiple-definitions` is attempting to compensate for a poor implementation of a COFF linker.  To make things worse, this is something which is hardcoded into the gcc specs IIRC, and has even partially made its way into clang :-(.  I for one welcome our new LLVM overlords^W^W^W^Wlld.